python/dgl/contrib/dis_kvstore.py

```diff
@@ -11,179 +11,106 @@
 from .._ffi.ndarray import empty_shared_mem
 
 import os
+import time
+import random
 import numpy as np
 import socket
+
 if os.name != 'nt':
     import fcntl
     import struct
 
+
+GARBAGE_COLLECTION_COUNT = 2000 # Perform grabage collection when message count is large than 2000
+
+
 def read_ip_config(filename):
-    """Read networking configuration from file.
+    """Read network configuration information of kvstore from file.
+
+    The format of configuration should be:
 
-    Format:
+        [machine_id] [ip] [base_port] [server_count]
 
-        [server_id] [ip] [port]
+        0 172.31.40.143 30050 2
+        1 172.31.36.140 30050 2
+        2 172.31.47.147 30050 2
+        3 172.31.30.180 30050 2
 
-        0 172.31.40.143 50050
-        1 172.31.36.140 50050
-        2 172.31.47.147 50050
-        3 172.31.30.180 50050
+    Note that, DGL KVStore supports backup servers that can shared data with each other 
+    servers on the same machine via shared-tensor, so the server_count should greater than 1. 
+    If server_count == 1, DGL luanches just 1 server node on a single machine.
 
     Parameters
     ----------
     filename : str
-        name of target configure file.
+        name of configuration file.
 
     Returns
     -------
     dict
-        server namebook, e.g., 
-
-          {0:'172.31.40.143:50050',
-           1:'172.31.36.140:50050',
-           2:'172.31.47.147:50050',
-           3:'172.31.30.180:50050'}
+        server namebook. e.g.,
+
+        [server_id]:[machine_id|server_addr]
```

Comment:
I think storing all the infomation in string will be easy for uses to wirte the configuration file. The machine_id is provided by user in configuration file and I think it will be useful when debugging.

Reply:
I will change the information of the server_namebook to a list.